### PR TITLE
Add destinations to server

### PR DESCRIPTION
**Pull request by steeevin88**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -6,6 +7,7 @@
 import typing
 from dataclasses import dataclass
 
+from datetime import datetime, timezone
```
Comment (evanugarte) [inline app.py]: lets import the whole thing instead
```suggestion
import datetime
```

Comment (steeevin88) [reply app.py]: resolved

File: app.py [hunk 2]
```diff
@@ -60,16 +63,28 @@ class Stop:
...
   predictions: typing.List[Prediction]
 
+@dataclass
+class Cache:
+  stops: typing.List[Stop]
+  updated_at: str
+
 PREDICTIONS_URL = 'https://api.511.org/transit/StopMonitoring'
 cache = []
+updated_cache = Cache([], '')
```
Comment (evanugarte) [inline app.py]: shouldnt it be
```suggestion
cache = []
```

File: app.py [hunk 3]
```diff
@@ -136,7 +146,7 @@ async def track_response_codes(request: Request, call_next):
 
 @app.get('/predictions')
 def predictions():
-    return cache
+    return updated_cache
```
Comment (evanugarte) [inline app.py]: ?
```suggestion
    return cache
```

File: app.py [hunk 4]
```diff
@@ -60,16 +63,28 @@ class Stop:
...
 
+@dataclass
+class Cache:
+  stops: typing.List[Stop]
+  updated_at: str
+
 PREDICTIONS_URL = 'https://api.511.org/transit/StopMonitoring'
 cache = []
+updated_cache = Cache([], '')
+time_updated = ''
```
Comment (NotMyPersonalAccount) [inline app.py]:
```suggestion
```

Unused

File: app.py (hunk 3 above)
Comment (steeevin88) [reply app.py]: resolved

File: app.py (hunk 4 above)
Comment (steeevin88) [reply app.py]: fixed

File: app.py (hunk 2 above)
Comment (steeevin88) [reply app.py]: resolved

File: app.py [hunk 5]
```diff
@@ -60,16 +63,26 @@ class Stop:
...
+        "BA": "BART",
+    }
+    new_cache = []
+
     # send post request for each agency's stop(s)
     for stop in stops:
         agency = stop.get('operator')
         stopCode = stop.get('id')
         stopName= stop.get('name')
+        if (agency in agency_to_stop_suffix):
```
Comment (evanugarte) [inline app.py:87]:
```suggestion
        # We add a suffix to some stop names to add context.
        # For example, the BART stop called "Milpitas" becomes
        # "Milpitas BART"
        if (agency in agency_to_stop_suffix):
```

File: app.py [hunk 6]
```diff
@@ -95,29 +108,23 @@ def update_cache():
...
-        unique_buses = {}
+        unique_buses: typing.Dict[str, Prediction] = collections.defaultdict(lambda: Prediction("", "", []))
         for bus in all_incoming_buses:
             route_name = bus.get('MonitoredVehicleJourney', {}).get('LineRef')
+            route_destination = bus.get('MonitoredVehicleJourney', {}).get('MonitoredCall', {}).get('DestinationDisplay').title()
             expected_arrival = bus.get('MonitoredVehicleJourney', {}).get('MonitoredCall', {}).get('AimedArrivalTime')
             
-            if route_name not in unique_buses:
-                unique_buses[route_name] = []
+            route_identifier = route_name + " to " + route_destination
```
Comment (evanugarte) [inline app.py]:
```suggestion
```

File: app.py [hunk 7]
```diff
@@ -95,29 +108,23 @@ def update_cache():
...
-            )
-            predictions.append(pred)
-
-        stopInfo = Stop(stopCode, stopName, predictions)
-        cache.append(stopInfo)
-        
-    return
+            unique_buses[route_identifier].route = route_name
+            unique_buses[route_identifier].destination = route_destination
+            unique_buses[route_identifier].times.append(expected_arrival)
```
Comment (evanugarte) [inline app.py]:
```suggestion
            unique_buses[route].route = route_name
            unique_buses[route].destination = route_destination
            unique_buses[route].times.append(expected_arrival)
```

File: app.py [hunk 8]
```diff
@@ -60,16 +63,26 @@ class Stop:
...
 PREDICTIONS_URL = 'https://api.511.org/transit/StopMonitoring'
-cache = []
+cache = Cache([], '')
 def update_cache():
-    # clear cache of any previous (now outdated) data
-    cache.clear()
+    agency_to_stop_suffix = {
+        "BA": "BART",
+    }
+    new_cache = []
```
Comment (NotMyPersonalAccount) [inline app.py]: `new_cache` poorly explains the use of this list, `new_stops` would probably be better suited

File: app.py (hunk 7 above)
Comment (NotMyPersonalAccount) [reply app.py]: `route` doesn't appear to be defined ?

Comment (evanugarte) [reply app.py]: my bad im fixing with him locally

File: app.py (hunk 8 above)
Comment (steeevin88) [reply app.py]: fixed

**Review by evanugarte — APPROVED** [review]

**Pull request merged**